contrib/utilities/download_clang_format

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2018 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE at
+## the top level of the deal.II distribution.
+##
+## ---------------------------------------------------------------------
+
+#
+# This script downloads and installs the clang-format binary. The
+# destination directory is
+#   [contrib/utilities]/programs/clang-<VERSION>/bin.
+#
+# This script only works on Linux (amd64) and macOS. For other
```

Comment:
Is this expected to work on `macOS` already? If so, we should include that in the OS check below.

Reply:
@masterleinad We will have macOS support once someone used the `compile_clang_format` script and sent me the result (*ping* @luca-heltai). I suggest we simply keep the comment in place and update the variant once I have created and uploaded a tarball.